<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Isuru HAUPE" />

<meta name="date" content="2018-08-01" />

<title>Fantasio</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />

</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 60px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 65px;
  margin-top: -65px;
}

.section h2 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h3 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h4 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h5 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h6 {
  padding-top: 65px;
  margin-top: -65px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Fantasio</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="gettingStarted.html">Getting Started</a>
</li>
<li>
  <a href="steps.html">Fantasio steps</a>
</li>
<li>
  <a href="references.html">Reference</a>
</li>
<li>
  <a href="https://github.com/genostats/FEstim">GitHub</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Fantasio</h1>
<h4 class="author"><em>Isuru HAUPE</em></h4>
<h4 class="date"><em>2018-08-01</em></h4>

</div>


<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>Fantasio is composed of several functions. Its goals are for:</p>
<ul>
<li>Population genetic studies (estimating and detecting inbreeding on individuals without known genealogy, estimating the population proportion of mating types and the individual probability to be offspring of different mating types)</li>
<li>Rare disease studies (performing homozygosity mapping with heterogeneity)</li>
<li>Multifactorial disease studies (HBD-GWAS strategy).</li>
</ul>
<p>Fantasio implements the creation of several random sparse submaps on genome-wide data (to remove linkage disequilibrium). It also provides graphical outputs to facilitate interpretations of homozygosity mapping results and plots.</p>
<p>In this vignette, we illustrate how to use the package, using the data set HGDP-CEPH, a ship which contains 1043 individuals and 660918 markers. In order to access the data, you will need to download the HGDP.CEPH package (see below how) and load it.</p>
<p>Not all options of the functions are described here, but rather their basic usage. The reader is advised to look at the manual page of the function for details.</p>
<div id="principal-concepts" class="section level3">
<h3>Principal concepts</h3>
<p>Fantasio implements a maximum likelihood method that uses a hidden Markov chain to model the dependencies along the genome between the (observed) marker genotypes of an individual, and its (unobserved) homozygous by descent (HBD) status. The emission probabilities of this hidden Markov model (HMM) depend on the allele frequencies. The transition probabilities depend on the genetic distance between two adjacent markers. This model allows estimating the inbreeding coefficient <span class="math inline">\(f\)</span> of an individual, and a parameter <span class="math inline">\(a\)</span>, where <span class="math inline">\(af\)</span> is the instantaneous rate of change per unit map length (here cM) from no HBD to HBD. Both HBD and non-HBD segment lengths are assumed to be distributed exponentially with mean lengths <span class="math inline">\(\frac{1}{a(1-f)}\)</span> and <span class="math inline">\(\frac{1}{af}\)</span>, respectively.</p>
<p>THe method requires the markers to be in minimal linkage disequilibrium (LD). Otherwise biased estimations of <span class="math inline">\(f\)</span> are produced. A strategy consisting of generating multiple random sparse genome maps (submaps) has been proposed to avoid this bias (Leutenegger et al. 2011). When several submaps are considered, <span class="math inline">\(f\)</span> is estimated as the median value on all the f estimation obtained on the different maps after removing submaps with <span class="math inline">\(a &gt; 1\)</span> (having an average HBD segment length of 1 cM is unlikely to be detected with a SNP density of 1 per 0.5 cM). This strategy has the advantage of not requiring any LD computation on the sample and of minimizing loss of information, as compared with a strategy that based on a single map of markers in minimal LD.</p>
<p>Fantasio also computes the likelihood of a mating type. These likelihoods can be used for:</p>
<ul>
<li>Inferring an individual as inbred by comparing the maximized likelihood with the one to be outbred with a likelihood ratio test</li>
<li>Estimating the population proportion of mating types</li>
<li>Estimating the individual probability to be into different mating types</li>
</ul>
<p>When multiple submaps are used, the median p-values/probabilities are considered. See Leutenegger et al. 2011 for more details on the calculations.</p>
<p>Homozygosity mapping (Lander and Botstein 1987) consists in focusing on inbred affected individuals and searching for a region of the genome of shared homozygosity. Original homozygosity mapping requires that the genealogy of patients be known so that inbred patients can be identified and their respective <span class="math inline">\(f\)</span> estimated. Leutenegger et al. (Leutenegger et al. 2006) proposed using the <span class="math inline">\(f\)</span> estimated on genome-wide genetic data to compute a FLOD score, similar to Mortons LOD score (Morton 1955).</p>
<p>Genin et al. (Genin et al. 2012) adapted the FLOD formula for multiple submaps. FLOD(i)(m,s) is computed for each individual <span class="math inline">\(i\)</span>, each marker <span class="math inline">\(m\)</span> on each submap <span class="math inline">\(s\)</span>, using the equation:</p>
<p><span class="math display">\[FLOD^{(i)}(m,s) = log_{10}\frac{P\left(Y_{m,s}^{(i)} | H_{1}\right)}{P\left(Y_{m,s}^{(i)} | H_{0}\right)} = log_{10}\frac{P\left(X_{m,s}^{(i)}=1 | Y_{m,s}^{(i)}\right) + q.P\left(X_{m,s}^{(i)}=0 | Y_{m,s}^{(i)}\right)}{\hat{f}_s^{(i)} + q.\left(1-\hat{f}_s^{(i)}\right)}\]</span></p>
<p>With the following parameters :</p>
<ul>
<li><span class="math inline">\(Y_{m,s}^{(i)}\)</span> the observed genotype of individual <em>i</em> at marker <em>m</em> on submap <em>s</em></li>
<li><span class="math inline">\(H_{1}\)</span> the hypothesis where <span class="math inline">\(Y_{m,s}^{(i)}\)</span> is linked to the disease, and <span class="math inline">\(H_{0}\)</span> the one where it is not</li>
<li><span class="math inline">\(X_{m,s}^{(i)}\)</span> the HBD status of individual <em>i</em> at marker <em>m</em> on submap <em>s</em> that is estimated together with the inbreeding coefficient using the HMM of the package</li>
<li><span class="math inline">\(\hat{f}_s^{(i)}\)</span> the estimated inbreeding coefficient of individual <em>i</em> on submap <em>s</em></li>
<li><em>q</em> the assumed frequency of the mutation involved in the disease for this individual.</li>
</ul>
<p>Results are then averaged over the different submaps to obtain a single <span class="math inline">\(FLOD^{(i)}(m)\)</span> at each marker <em>m</em>.</p>
<p>Genin et al. (Genin et al. 2012) proposed to detect fully penetrant rare recessive variants by performing homozygosity mapping on inbred cases from Genome-Wide Association Study (GWAS) data. Linkage evidence is then evaluated over the entire set I of inbred cases by computing a FLOD score, HFLOD(m,<span class="math inline">\(\alpha\)</span>), at each marker m, with a heterogeneity parameter <span class="math inline">\(\alpha\)</span>, that takes into account the possibility that only a fraction of the inbred affected individuals carry diseases causing mutations:</p>
<p><span class="math display">\[HFLOD(m,\alpha)=\sum log_{10} \left[\alpha.\frac{P\left(Y_{m,s}^{(i)} | H_{1}\right)}{P\left(Y_{m,s}^{(i)} | H_{0}\right)}+ (1 - \alpha)\right ]= \sum log_{10} \left[\alpha . exp \left(FLOD^{(i)}(m)*log(10)\right)+(1-\alpha)\right] \]</span> This heterogeneity score is then maximized over <span class="math inline">\(\alpha\)</span> to evaluate the evidence of linkage at marker <em>m</em> where <span class="math inline">\(\alpha\)</span> is the estimate of the proportion of cases linked to this locus:</p>
<p><span class="math display">\[HFLOD(m)=max_{\alpha}(HFLOD(m,\alpha))\]</span></p>
</div>
</div>
<div id="installing-festim" class="section level1">
<h1>Installing FEstim</h1>
<p>In your R terminal type :</p>
<pre class="r"><code>install.packages(&quot;Fantasio&quot;)</code></pre>
<p>If you need to plot using plinkâ€™s ROH files output, make PLINK (Purcell et al. 2007) is installed, we recommend the quicker version from the BGI, <a href="https://www.cog-genomics.org/plink2/" target="_blank">plink2</a>.</p>
</div>
<div id="citing-festim" class="section level1">
<h1>Citing FEstim</h1>
<ul>
<li><p>When using multiple submaps: Leutenegger AL, Sahbatou M, Gazal S, Cann H, Genin E. 2011. Consanguinity around the world: what do the genomic data of the HGDP-CEPH diversity panel tell us? Eur J Hum Genet 19: 583-587.</p></li>
<li>When estimating f: Leutenegger AL, Prum B, Genin E, Verny C, Lemainque A, Clerget-Darpoux F, Thompson EA.</li>
</ul>
<ol start="2003" style="list-style-type: decimal">
<li>Estimation of the inbreeding coefficient through use of genomic data. Am J Hum Genet 73: 516-523.</li>
</ol>
<ul>
<li><p>When calculating FLOD scores: Leutenegger AL, Labalme A, Genin E, Toutain A, Steichen E, Clerget-Darpoux F, Edery P. 2006. Using genomic inbreeding coefficient estimates for homozygosity mapping of rare recessive traits: application to Taybi-Linder syndrome. Am J Hum Genet 79: 62-66.</p></li>
<li><p>When performing HBD-GWAS strategy: Genin E, Sahbatou M, Gazal S, Babron MC, Perdry H, Leutenegger AL. 2012. Could Inbred Cases Identified in GWAS Data Succeed in Detecting Rare Recessive Variants Where Affected Sib-Pairs Have Failed? Hum Hered 74: 142-152.</p></li>
</ul>
</div>
<div id="reporting-problems-bugs-and-questions" class="section level1">
<h1>Reporting problems, bugs and questions</h1>
<p>If you have any problems or suggestions with this package, please contact Mr Herve PERDRY : <a href="mailto:herve.perdry@u-psud.fr">herve.perdry@u-psud.fr</a></p>
</div>

<p>Copyright &copy; 2018 INSERM. All rights reserved.</p>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
